lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -455,7 +495,21 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
       return false;
     }
   }
-
+  // Converged 5G rules to install
+  for (const auto& rule : uc.PdrList_to_install) {
+    if (!search_pdr(rule.pdr_id())) {
+      int32_t Pdr_index;
+      Pdr_index = pdr_index(rule.pdr_id());
```

Comment:
effectively you are performing the search twice. What is the intent here? On line 500, you are checking if the pdr id exists and if not you are skipping, then on 503 you are again checking whether the id exists or not and then update it on line 505.

Reply:
Changed the logic.